File: src/main/java/com/koddy/server/member/presentation/dto/request/MentoringPeriodRequest.java
package com.koddy.server.member.presentation.dto.request;

import com.koddy.server.member.domain.model.mentor.MentoringPeriod;

import java.time.LocalDate;

public record MentoringPeriodRequest(
        LocalDate startDate,
        LocalDate endDate
) {
    public MentoringPeriod toPeriod() {
        if (startDate != null && endDate != null) {
            return MentoringPeriod.of(startDate, endDate);
        }
        return null;
    }
}

File: src/main/java/com/koddy/server/member/presentation/dto/request/MentoringPeriodRequest.kt
package com.koddy.server.member.presentation.dto.request

import com.koddy.server.member.domain.model.mentor.MentoringPeriod
import java.time.LocalDate

data class MentoringPeriodRequest(
    val startDate: LocalDate?,
    val endDate: LocalDate?,
) {
    fun toPeriod(): MentoringPeriod? {
        if (startDate != null && endDate != null) {
            return MentoringPeriod.of(startDate, endDate)
        }
        return null
    }
}
